expression/expression.go

```diff
@@ -243,17 +243,42 @@ func VecEvalBool(ctx sessionctx.Context, exprList CNFExprs, input *chunk.Chunk,
 			return nil, nil, err
 		}
 
+		j := 0
+		if eType == types.ETInt {
```

Comment:
Could we do the same thing for `ETReal`, `ETDecimal`, ...?

Reply:
In the previous version, only the int type had special handling, see here https://github.com/pingcap/tidb/blob/master/expression/chunk_executor.go#L373. Other types can be solved by [```vecEval```](https://github.com/pingcap/tidb/blob/master/expression/expression.go#L312).

Reply:
`vecEval()` converts the result into `Datum` then calculate the bool result. It's inefficient. I think it might much faster if we directly handle the float, decimal, datetime, ... result to calculate the bool result.

Reply:
Do you mean by directly judging the type to do different processing, rather than by calling ```vecEval()```?

Reply:
Yeah, exactly. Just like what we did for `ETInt` in this PR.

Reply:
The purpose of this PR is to implement the vectorization of the selection operator without rollback performance. Maybe we can support judging type directly instead of datum conversion in the next PR. What's your opinion? @zz-jason

Reply:
But you already did part of the optimizations for `ETInt`?